src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1356,58 +1371,69 @@ class JoinClauseSegment(BaseSegment):
     """Any number of join clauses, including the `JOIN` keyword."""
 
     type = "join_clause"
-    match_grammar = Sequence(
+    match_grammar = OneOf(
         # NB These qualifiers are optional
         # TODO: Allow nested joins like:
         # ....FROM S1.T1 t1 LEFT JOIN ( S2.T2 t2 JOIN S3.T3 t3 ON t2.col1=t3.col1) ON
         # tab1.col1 = tab2.col1
-        OneOf(
-            "CROSS",
-            "INNER",
-            Sequence(
-                OneOf(
-                    "FULL",
-                    "LEFT",
-                    "RIGHT",
-                ),
-                Ref.keyword("OUTER", optional=True),
-            ),
-            optional=True,
-        ),
-        Ref("JoinKeywords"),
-        Indent,
         Sequence(
```

Comment:
Use Hide Whitespace to review this and it'll look at lot smaller as this section was indented to allow `OneOf` above for the NATURAL join section below.

![image](https://user-images.githubusercontent.com/10931297/151597315-e5db2f6b-fb1e-4724-868f-b17ad163cfd8.png)